### DISCUSSION TITLE
nx workspace-lint

### DISCUSSION THREAD

**Discussion by ronderksen** [nrwl/nx#5278] [General]
What exactly does nx workspace-lint do? Does it check the nx.json, workspace.json and files like that? The documentation doesn't really clarify that for me.

**Comment by meeroslav** [comment]
It checks the integrity of `workspace.json` and `nx.json` and checks whether all files are existing according to `workspace.json` and `nx.json`.

For example, if you would manually rename app folder, but leave config files intact, it would report that app can no longer be found at the location specified.

**Reply by theerebuss** [reply]
I also went through this question when looking at the doc information. I think it'd be clearer to use something like:
`Lint nx specific workspace files (nx.json, workspace.json) and check existence of the configured packages and apps.`

Would a PR with this specific change be welcome?

**Reply by meeroslav** [reply]
Sure, please propose a PR that improves this.

**Comment by skoblenick** [comment]
Related to the improved documentation... I would question how I configure workspace-lint. As an example, I personally want to add `README.md` files to my project structure, for documentation purposes, and because they exist within in the `./apps/`, `./apps/<scope>/`, or `/apps/<scope>/<scope>` folders the `nx workspace-lint` throws errors for files it shouldn't really be concerned with; especially as some of these fall outside of the scope of the applications/libraries. For example:

```
> nx "workspace-lint"

>  NX   ERROR  The following file(s) do not belong to any projects:

  - apps/team1/mfe1/README.md
  - apps/team1/mfe2/README.md
  - apps/team1/README.md
  - apps/README.md
```

With a structure like:

```
apps
   team1/
      mfe1/
          api
          web
          web-e2e
          README.md
       mfe2/
          api
          web
          web-e2e
          README.md
       README.md
   README.md
```

**Reply by skoblenick** [reply]
As a solution to anyone looking for a workaround, see https://github.com/nrwl/nx/issues/1717. In summary add `README.md` to a `.nxignore`
